speasy/core/codecs/bundled_codecs/hapi_csv/reader.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional, Union, Dict, Any, Tuple
+import io
+import logging
+
+import numpy as np
+import pandas as pds
+import json
+
+from speasy.core.codecs.codec_interface import Buffer
+from speasy.core.any_files import any_loc_open
+from .csv_file import HapiCsvFile
+
+log = logging.getLogger(__name__)
+
+
+def _extract_headers(file: io.IOBase) -> Dict[str, Any]:
+    headers = []
+    line = file.readline()
+    while line:
+        if line.startswith('#'):
+            headers.append(line[1:].strip())
+        else:
+            break
+        line = file.readline()
+    if len(headers):
+        return json.loads(''.join(headers))
+    return {}
+
+
+def _extract_data(file: io.IOBase) -> pds.DataFrame:
+    file.seek(0)
+    return pds.read_csv(file, comment='#', header=None, skiprows=0)
```

Comment:
With Python 3.12.7 and pandas 2.2.3:

```python
import pandas as pds

df = pds.read_csv("tests/resources/HAPI_sample_csv.csv", comment='#', header=None, skiprows=0)

print(df.index.values)
```
The result is:
`array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47])`

=> Dataframe index is not the "time" column and there is something wrong with (cf. line 55 in load_hapi_csv method): 
```python
hapi_csv_file.create_parameter(data.index.to_numpy(dtype="datetime64[ns]"), meta=headers["parameters"][0])
```

Reply:
```python
import speasy as spz
from speasy.core.codecs import get_codec

hapi_csv_codec = get_codec('hapi/csv')

v = hapi_csv_codec.load_variable("Magnitude", "tests/resources/HAPI_sample_csv.csv", disable_cache=True)

print(v.time)
```